cirq/ion/ion_device.py

```diff
@@ -186,7 +185,7 @@ def _value_equality_values_(self):
 def _verify_unique_measurement_keys(operations: Iterable[ops.Operation]):
     seen: Set[str] = set()
     for op in operations:
-        meas = ops.op_gate_of_type(op, ops.MeasurementGate)
+        meas = op.gate if isinstance(op.gate, ops.MeasurementGate) else None
         if meas:
```

Comment:
The inline if can be merged with the if statement.

Reply:
Agreed, looks much better to put isinstance in the if condition. Changed throughout. (This sort of thing will be even nicer with the walrus operator in 3.8.)